core-backend/sandbox/src/runtime.rs

```diff
@@ -92,80 +99,140 @@ impl<Ext: BackendExternalities> CommonRuntime<Ext> for Runtime<Ext> {
         F: FnOnce(&mut Self) -> Result<T, RunFallibleError>,
         R: From<Result<T, u32>> + Sized,
     {
-        self.run_any(gas, cost, |ctx| {
-            let res = f(ctx);
-            let res = ctx.process_fallible_func_result(res)?;
-
-            // TODO: move above or make normal process memory access.
-            let write_res = ctx.memory_manager.register_write_as::<R>(res_ptr);
-
-            ctx.write_as(write_res, R::from(res)).map_err(Into::into)
-        })
+        self.run_any(
+            gas,
+            cost,
+            |ctx: &mut Self| -> Result<_, UndefinedTerminationReason> {
+                let res = f(ctx);
+                let res = ctx.host_state_mut().process_fallible_func_result(res)?;
+
+                // TODO: move above or make normal process memory access.
```

Comment:
yep, please resolve. Memory accesses should be registered before the function call

Reply:
why should it be resolved? @gshep

Reply:
This is pre_process_memory_access ineffective work problem. Will be solved one day))